etc/vendor-libmongoc.sh

```diff
@@ -3,8 +3,14 @@
 set -eou pipefail
 
 PWD=`pwd`
-LIBMONGOC_VERSION=1.18.0
-TARBALL_URL=https://github.com/mongodb/mongo-c-driver/releases/download/$LIBMONGOC_VERSION/mongo-c-driver-$LIBMONGOC_VERSION.tar.gz
+LIBMONGOC_MAJOR_VERSION=1
+LIBMONGOC_MINOR_VERSION=18
+LIBMONGOC_PATCH_VERSION=0
+LIBMONGOC_PRERELEASE_VERSION=
```

Comment:
I tested the logic for when a prerelease version is specified here by setting it to "alpha2", as 1.18.0-alpha2 was a recent prerelease.
In that case script fails toward the end at the stage where we apply the various .diff files due to incompatibilities with those (when we vendored in 1.18.0 I had to update the patch files a bit), but the new logic added here works anyway.